### PR TITLE
Automate production deployments

### PR DESCRIPTION
**Pull request by jeffmaher**
Adds a deployment script and GitHub Actions run step for when a branch is merged to `main`. Does a few things in the script:

1. Checkout the highest tag version
2. Sets up Terraform (to sync with the server-side stored state)
3. Sees if there are any infrastructure changes and makes those changes
4. Builds and deploys an updated version of the app

Also updates the `README.md` with revised instructions, particularly for manual deployments in the event of a failure or deploying a dev environment.

**Know issues** (review question: are these worth fixing ATM?):

There's a slight, but unlikely, race condition. If there are rapid fire merges, this can fail for two reasons:

- Terraform will be locked if multiple deployments are happening
- This runs the job in the `main` Dockerfile and deploys code from highest tag number (i.e. likely to be the same). But then it might have a stale or slightly newer `Dockerfile` if a merge rapidly follows

### CODE REVIEW

**Comment by jeffmaher** [comment]
NOT READY FOR REVIEW YET: Testing the job and script

**Comment by jeffmaher** [comment]
Copied the run command into the pr_checks to test the command and script (via PR change events, since merge events are harder to test rapidly), and got it working:

image

Removed the test line now from `pull_request.yml` and leaving in `tag_main.yml`

**Comment by jeffmaher** [comment]
Updated GitHub Actions Secrets with production values (changed from my dev ones).

**Comment by jeffmaher** [comment]
@bloom-works/co-bha-care-directory-devs : Ready for review!

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -44,4 +44,4 @@ jobs:
 
       # CODE QUALITY
       - name: Run code quality scans
-        run: docker-compose -f docker-compose.yml -f ci/docker-compose.ci.yml run app sh ci/code_quality.sh
\ No newline at end of file
+        run: docker-compose -f docker-compose.yml -f ci/docker-compose.ci.yml run app sh ci/code_quality.sh
```
Comment (jeffmaher) [inline .github/workflows/pull_request.yml:47]: No changes here -- just a newline

**Comment by jeffmaher** [comment]
Whoops -- have a syntax bug. Working on:
https://github.com/bloom-works/co-care-directory/actions/runs/2674069116/workflow

File: ci/publish_build.sh [hunk 2]
```diff
@@ -4,7 +4,31 @@
...
+# Variables
+APP_DIR="/app"
+INFRA_DIR="$APP_DIR/infra/aws/infra"
+
+# Checkout the new tag
+cd $APP_DIR
+git checkout $1
+
+# Initialize Terraform
+# docker run -v "$PWD:/app" -w "$INFRA_DIR" -e TF_VAR_bucket_name -e TF_VAR_domains -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY coloradodigitalservice/co-care-directory terraform init -backend-config="bucket=${TF_VAR_bucket_name}-terraform-state" -backend-config="dynamodb_table=${TF_VAR_bucket_name}-terraform-state"
```
Comment (jlhogan) [inline ci/publish_build.sh:16]: these commented out `docker run` commands are for running these steps locally ? Should they stay in this file? if yes, can they be contextualized via a comment at top explaining what they are/why they are included

File: .github/workflows/tag_main.yml [hunk 3]
```diff
@@ -26,7 +26,12 @@ jobs:
...
-
- 
+      - name: Build Docker container
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          load: true
+          tags: ${{ github.repository }}:${{ github.head_ref }}
+      - name: Build and publish artifacts
```
Comment (jlhogan) [inline .github/workflows/tag_main.yml:36]: Publish to prod only? What about staging?

File: ci/publish_build.sh (hunk 2 above)
Comment (jeffmaher) [reply ci/publish_build.sh:16]: Good catch! Will remove those.

File: .github/workflows/tag_main.yml (hunk 3 above)
Comment (jeffmaher) [reply .github/workflows/tag_main.yml:36]: Yep! CO was fine doing pushes to prod directly from here, continuous delivery style -- assuming we're doing reviews and getting approvals in PRs post-hard launch.

Comment (jlhogan) [reply .github/workflows/tag_main.yml:36]: OK, in that case I think we should also deploy to staging as part of this process to ensure it remains up to date. maybe make the deploys sequential as a very minimal assurance that the deploy isn't borked

**Pull request merged**